Added sample on how to use Solace OpenTelemetry Integration API

NeeloferS opened the pull request:
Sample to show:

- adding trace context to Solace message before publishing
- extracting trace context from Solace message after receiving
- adding baggage to Solace message
- extracting baggage from Solace message

NeeloferS committed:
Added sample on how to use SolaceMessageCarrier.Setter and SolaceMessageCarrier.Getter

NeeloferS committed:
fixed directory structure

TamimiGitHub commented on src/TracingManualInstrumentation/TracingManualInstrumentation.cs:
```diff
@@ -0,0 +1,374 @@
+                    {
+                        // Create the message content as a binary attachment
+                        message.BinaryAttachment = Encoding.ASCII.GetBytes(
+                            string.Format("Tracing Manual Instrumentation Sample! Message ID: {0}", i));
```
If I got this correct, this is not the message ID yeah? I would suggest changing this to
`("Message number: {0}", i)`

TamimiGitHub commented on src/TracingManualInstrumentation/TracingManualInstrumentation.cs:
```diff
@@ -0,0 +1,374 @@
+                            string.Format("Tracing Manual Instrumentation Sample! Message ID: {0}", i));
+
+                        // Send the message to the queue on the Solace messaging router
+                        Console.WriteLine("Sending message ID {0} to topic '{1}' mapped to queue '{2}'...",
```
I would change the console log to 
> Sending messaging number '{0}' on topic '{1}' mapped to queue '{2}'

TamimiGitHub commented on src/TracingManualInstrumentation/TracingManualInstrumentation.cs:
```diff
@@ -0,0 +1,374 @@
+            using var activity = activitySource.StartActivity(activityName, ActivityKind.Producer);
+            activity?.SetTag("messaging.system", "Solace Pubsub plus");
+            activity?.SetTag("messaging.destination_kind", "queue");
+            activity?.SetTag("messaging.destination", "queueName");
```
I would suggest changing these two tags:
```
activity?.SetTag("messaging.destination_kind", "queue");
activity?.SetTag("messaging.destination", "queueName");
```
to:
```
activity?.SetTag("messaging.message_delivery_mode", "persistent");
activity?.SetTag("messaging.destination", "topic");
```

Mrc0113 replied:
Can we also make sure we use `solace` for the messaging.system since that is what we already have in the JCSMP samples. 

@TamimiGitHub - FYI here. we will probably want to double check other samples and make sure we follow this same example whenever we create content

NeeloferS committed:
- Fixed tag messaging.system value to solace
- Changed "Sending messaging number '{0}' to topic " to "...on topic"
- Fixed message ID to message number in binary attachment of message

NeeloferS committed:
Update TracingManualInstrumentation.cs

changed typo message ID to message number and message.destination to topic
